pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/juniperkits/JuniperKitCreationStatusTest.java

```diff
@@ -295,30 +296,33 @@ public void invalidKitTypeTest() {
-        String json = "{ \"firstName\":\"P\","
-                + "\"lastName\":\"T\","
-                + "\"street1\":\"415 Main st\","
-                + "\"street2\":null,"
-                + "\"city\":\"Cambridge\","
-                + "\"state\":\"MA\","
-                + "\"postalCode\":\"02142\","
-                + "\"country\":\"USA\","
-                + "\"phoneNumber\":\" 111 - 222 - 3344\","
-                + "\"juniperKitId\":\"JuniperTestKitId_" + rand + "\","
-                + "\"juniperParticipantID\":\"" + participantId + rand + "\","
-                + "\"skipAddressValidation\":false,"
-                + "\"juniperStudyID\":\"Juniper-test-guid\"}";
-
-        JuniperKitRequest juniperTestKit = new Gson().fromJson(json, JuniperKitRequest.class);
-        createdKitIds.add(juniperTestKit.getJuniperKitId());
-        KitResponse kitResponse =
-                nonPepperKitCreationService.createNonPepperKit(juniperTestKit, kitType, mockEasyPostUtil, ddpInstance);
-        Assert.assertEquals(KitResponse.ErrorMessage.UNKNOWN_KIT_TYPE, kitResponse.getErrorMessage());
-        Assert.assertEquals(kitResponse.getValue(), kitType);
-        List<KitRequestShipping> newKits = KitRequestShipping.getKitRequestsByRealm(instanceName, "overview", kitType);
-        Assert.assertEquals(newKits.size(), oldkits.size());
-
+        try {
+            KitRequestShipping.getKitRequestsByRealm(instanceName, "overview", kitType);
+        } catch (DsmInternalError e) {
+            Assert.assertTrue(e.getMessage().contains(String.format("No kit was found for study %s with kit type name %s", instanceName, kitType)));
+            String json = "{ \"firstName\":\"P\","
+                    + "\"lastName\":\"T\","
+                    + "\"street1\":\"415 Main st\","
+                    + "\"street2\":null,"
+                    + "\"city\":\"Cambridge\","
+                    + "\"state\":\"MA\","
+                    + "\"postalCode\":\"02142\","
+                    + "\"country\":\"USA\","
+                    + "\"phoneNumber\":\" 111 - 222 - 3344\","
+                    + "\"juniperKitId\":\"JuniperTestKitId_" + rand + "\","
+                    + "\"juniperParticipantID\":\"" + participantId + rand + "\","
+                    + "\"skipAddressValidation\":false,"
+                    + "\"juniperStudyID\":\"Juniper-test-guid\"}";
+
+            JuniperKitRequest juniperTestKit = new Gson().fromJson(json, JuniperKitRequest.class);
+            createdKitIds.add(juniperTestKit.getJuniperKitId());
+            KitResponse kitResponse =
+                    nonPepperKitCreationService.createNonPepperKit(juniperTestKit, kitType, mockEasyPostUtil, ddpInstance);
+            Assert.assertEquals(KitResponse.ErrorMessage.UNKNOWN_KIT_TYPE, kitResponse.getErrorMessage());
+            Assert.assertEquals(kitResponse.getValue(), kitType);
+            return;
+        }
+        Assert.fail();
```

Comment:
This fail should be in the try block above, and should add a message about the failure.